src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -135,15 +135,17 @@ void GcsAutoscalerStateManager::GetPendingGangResourceRequests(
   // Iterate through each placement group load.
   for (const auto &pg_data : placement_group_load->placement_group_data()) {
     auto gang_resource_req = state->add_pending_gang_resource_requests();
+    auto pg_state = pg_data.state();
+    auto pg_id = PlacementGroupID::FromBinary(pg_data.placement_group_id());
     // For each placement group, if it's not pending/rescheduling, skip it since.
     // it's not part of the load.
-    RAY_CHECK(pg_data.state() == rpc::PlacementGroupTableData::PENDING ||
-              pg_data.state() == rpc::PlacementGroupTableData::RESCHEDULING)
-        << "Placement group load should only include pending/rescheduling PGs. ";
+    if (pg_state != rpc::PlacementGroupTableData::PENDING &&
```

Comment:
if we are not adding removed pg's, do we still need to remove this check now?

Reply:
Yeah I think technically it is not necessary... but since it is hard to test this regression, I would just keep it here.

Reply:
i am adding a bunch of metrics to this - and I think i could surface this through that maybe later on instead of simply continuing.